### ISSUE TITLE
Make ROOT_DATA_PATH configurable

### ISSUE THREAD

**Issue by mariospr** [learningequality/ka-lite#5143]
## Summary

While trying to bundle KA Lite as a [flatpak bundle](http://flatpak.org) I realized that setting the `KALITE_DIR` variable to point to the alternative root directory (`/app/share/kalite`, for the case of flatpak) was not enough, since `kalite start` would fail with an error on startup, since it would still look into `/usr/share/kalite` when copying the static media files.

By looking into [kalite/__init__.py](https://github.com/learningequality/ka-lite/blob/master/kalite/__init__.py), it seems the problem is that the `ROOT_DATA_PATH` is always using python's `sys.prefix` to construct the data path, which will point to the non-existent `/usr/share/kalite` in this particular case.

So, even though I agree one option to prevent this could be to avoid doing that copy of static media, I think it would be nice if we could construct the `ROOT_DATA_PATH` variable considering the value of the `KALITE_DIR` variable, if it exists.

## Branch or installer method

I'm using a development environment setup on top of my Fedora 23 machine using virtualenv and  [this instructions](https://github.com/learningequality/ka-lite/wiki/Getting-started), with the **`0.16.x` branch**.

## Traceback or relevant snippet from server.log

On first run of `kalite start` with KA lite installed in a non-standard location (`/app/share/kalite` for flatpak), it will fail with this error:
```
[...]
[INFO] [2016-06-08 15:23:40,480] root: Copying static media...
Traceback (most recent call last):
  File "/app/share/kalite/python-packages/django/core/management/base.py", line 224, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/app/share/kalite/python-packages/django/core/management/base.py", line 263, in execute
    output = self.handle(*args, **options)
  File "/app/lib/python2.7/dist-packages/kalite/distributed/management/commands/initialize_kalite.py", line 42, in handle
    self.setup_server_if_needed()
  File "/app/lib/python2.7/dist-packages/kalite/distributed/management/commands/initialize_kalite.py", line 36, in setup_server_if_needed
    call_command("setup", interactive=False)
  File "/app/share/kalite/python-packages/django/core/management/__init__.py", line 161, in call_command
    return klass.execute(*args, **defaults)
  File "/app/share/kalite/python-packages/django/core/management/base.py", line 263, in execute
    output = self.handle(*args, **options)
  File "/app/lib/python2.7/dist-packages/kalite/distributed/management/commands/setup.py", line 494, in handle
    clear=True)
  File "/app/share/kalite/python-packages/django/core/management/__init__.py", line 161, in call_command
    return klass.execute(*args, **defaults)
  File "/app/share/kalite/python-packages/django/core/management/base.py", line 263, in execute
    output = self.handle(*args, **options)
  File "/app/share/kalite/python-packages/django/core/management/base.py", line 393, in handle
    return self.handle_noargs(**options)
  File "/app/share/kalite/python-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 164, in handle_noargs
    collected = self.collect()
  File "/app/share/kalite/python-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 105, in collect
    for path, storage in finder.list(self.ignore_patterns):
  File "/app/share/kalite/python-packages/django/contrib/staticfiles/finders.py", line 106, in list
    for path in utils.get_files(storage, ignore_patterns):
  File "/app/share/kalite/python-packages/django/contrib/staticfiles/utils.py", line 25, in get_files
    directories, files = storage.listdir(location)
  File "/app/share/kalite/python-packages/django/core/files/storage.py", line 248, in listdir
    for entry in os.listdir(path):
OSError: [Errno 2] No such file or directory: '/usr/share/kalite/static-libraries'
```

## How to reproduce

I reproduce this after bundling the KA Lite server inside a flatpak but I believe an alternative reliable way to reproduce it should be:

1. Install KA Lite to a non-standard location (other than `<sys.prefix>/share/kalite` 
2. Make sure KA Lite has not been run before, or delete previous data in `CONTENT_ROOT`:
```
  $ rm -rf ~/.kalite
``` 
3. Run `kalite start --foreground` for the first time and observe the error on command line
4. Connect to the web UI from localhost:8008 and observe how several UI assets are not loaded (and multiple 404 errors on the terminal)

## Real-life consequences

Probably not a life-or-death thing, but I still believe it would be a nice thing to have as it makes easier to run KA Lite in certain non-standard environments, such as inside of a flatpak bundle

**Comment by mariospr** [comment]
Just pushed a pull request which I hope will be fine, although I'm unsure in how to add a test for it.

Any comment/help welcome. Thanks!

**Comment by benjaoming** [comment]
@mariospr the user data directory `KALITE_DIR` and `sys.prefix`-based directory `ROOT_DATA_PATH` are two different things. While the latter would be okay to have configurable, it does not make sense to have it respect the `KALITE_DIR` as it can be set while still wanting access to the system-wide directory.

I would propose that you fix it by introducing a separate `KALITE_SYS_PREFIX` setting.

**Comment by benjaoming** [comment]
Thanks for opening up the issue and explaining the motivation btw :)

**Comment by mariospr** [comment]
Ah! I knew I was missing something... that (a new variable) was actually my initial approach, but today I came across this again and thought it would make sense to make it respect KALITE_DIR.

I'll try this new idea now (already renamed the issue). Thanks!

**Comment by mariospr** [comment]
On a second thought, after looking into this closer and asking on IRC to @benjaoming, it is unclear to me that splitting it to a new environment variable is the right thing to do, since KALITE_SYS_PREFIX/share/kalite should probably match KALITE_DIR anyway.

Talking to @benjaoming on IRC it seems the source of the misunderstanding could come from confusing the KALITE_DIR variable, which should point to the root of the installation (under the system prefix) with the KALITE_HOME variable, which should point to where the user data files are stored.

I have prepared an alternative PR locally introducing the new variable, but before changing the PR I'd appreciate some confirmation, since it might be that the first approach was right after all? :)

**Comment by mariospr** [comment]
JFTR, we agreed on the PR to go with the initial approach, now pending on final review. Thanks!

**Comment by benjaoming** [comment]
Fixed in #5144
